src/k2/dto/Collection.cpp

```diff
@@ -135,20 +141,27 @@ PartitionGetter::PartitionWithEndpoint& PartitionGetter::getPartitionForKey(cons
             throw std::runtime_error("no partition for endpoint");
         }
         default:
-            throw std::runtime_error("no partition for endpoint");
+            throw std::runtime_error("Unknown hash scheme for collection");
     }
 }
 
 OwnerPartition::OwnerPartition(Partition&& part, HashScheme scheme) :
     _partition(std::move(part)),
-    _scheme(scheme),
-    _hstart(std::stoull(_partition.startKey)),
-    _hend(std::stoull(_partition.endKey)) {
+    _scheme(scheme) {
+
+    if (_scheme == HashScheme::HashCRC32C) {
+        _hstart = std::stoull(_partition.startKey);
+        _hend = std::stoull(_partition.endKey);
+    }
 }
 
 bool OwnerPartition::owns(const Key& key) const {
     switch (_scheme) {
         case HashScheme::Range:
+            if (_partition.endKey == "") {
+                return _partition.startKey.compare(key.partitionKey) <= 0;
+            }
+            // TODO should endKey comparision be strictly less than?
```

Comment:
I think so, yes. Otherwise we can't express two partitions to be [""..."b", "b"..""]. We need the key "b" to not be in the first partition in this case - it should be in the second. In other words, partitions should be [start; end)
We should also revisit hash-based partitioning to make sure it follows the same convention